tools/json_schemas_to_typescript.rb

```diff
@@ -21,64 +21,116 @@ def initialize_listener(app)
 
       app.config.after_initialize do
         listener =
-          ::Listen.to(Rails.root.join(SCHEMA_DIRECTORY).to_s) do |_changed, _added, _removed|
-            JsonSchemasToTypescript.write_files
+          ::Listen.to(Rails.root.join(SCHEMA_DIRECTORY).to_s) do |changed, added, removed|
+            JsonSchemasToTypescript.write_files(
+              changed: changed.map { relative_path(it) },
+              added: added.map { relative_path(it) },
+              removed: removed.map { relative_path(it) },
+            )
```

Comment:
Type compilation is much slower now, so rather than just always generating all types, we now want to be selective, and only generate the necessary types, depending on what has changed.